ADAL/src/cache/ADTokenCacheAccessor.m

```diff
@@ -332,6 +335,20 @@ - (void)updateCacheToItem:(ADTokenCacheItem *)cacheItem
     [[ADTelemetry sharedInstance] stopEvent:telemetryRequestId event:event];
 }
 
+- (BOOL)addOrUpdateItem:(nonnull ADTokenCacheItem *)item
+                context:(id<ADRequestContext>)context
+                  error:(ADAuthenticationError * __nullable __autoreleasing * __nullable)error
+{
+    NSURL *oldAuthority = [NSURL URLWithString:item.authority];
+    NSURL *newAuthority = [[ADAuthorityValidation sharedInstance] cacheUrlForAuthority:oldAuthority context:context];
+    
+    item.authority = [newAuthority absoluteString];
+    BOOL ret = [_dataSource addOrUpdateItem:item correlationId:context.correlationId error:error];
+    item.authority = [oldAuthority absoluteString];
```

Comment:
nit: please add a comment here about, because it's difficult to understand without knowing context

Reply:
👍